neo/Consensus/ConsensusService.cs

```diff
@@ -219,12 +226,16 @@ private void OnCommitReceived(ConsensusPayload payload, Commit commit)
             {
                 Log($"{nameof(OnCommitReceived)}: height={payload.BlockIndex} view={commit.ViewNumber} index={payload.ValidatorIndex} nc={context.CountCommitted()} nf={context.CountFailed()}");
 
-                byte[] hashData = context.MakeHeader()?.GetHashData();
-                if (hashData == null)
+                byte[] hashData1 = context.MakeHeader()?.GetHashData();
+                byte[] hashData2 = context.MakeStateRoot()?.GetHashData();
+                if (hashData1 == null || hashData2 == null)
```

Comment:
Signature of both hashes concatenated?

Reply:
It can't, one signature is for block, another is for state root mssage, as block and state root are two different messages.